pkg/sops/sops.go

```diff
@@ -0,0 +1,88 @@
+package sops
+
+import (
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/getsops/sops/v3"
+	"github.com/getsops/sops/v3/stores/dotenv"
+	"github.com/getsops/sops/v3/stores/ini"
+	"github.com/getsops/sops/v3/stores/json"
+	"github.com/getsops/sops/v3/stores/yaml"
+)
+
+func getStore(fileType string) (sops.Store, error) {
+	switch fileType {
+	case ".ini":
+		return &ini.Store{}, nil
+	case ".env":
+		return &dotenv.Store{}, nil
+	case ".yaml", ".yml":
+		return &yaml.Store{}, nil
+	case ".json":
+		return &json.Store{}, nil
+	default:
+		return nil, fmt.Errorf("Unsupported file type: %s", fileType)
+	}
+}
+
+// File represents a SOPS file and its metadata
+type File struct {
+	Path     string
+	Metadata sops.Metadata
+}
+
+// FindFiles searches a directory for YAML files and checks if they are valid SOPS files.
+// Returns []SopsFile
+func FindFiles(root string) ([]File, error) {
+	var SopsFiles []File
```

Comment:
Variable names should use camelCase:

```suggestion
	var sopsFiles []File
```